### PR TITLE
feat: Show more numbers in human readable form

### PR DESCRIPTION
**Pull request by flub**
This shows sizes in KiB, MiB etc.  Also adds a few totals and transfer
summaries.

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]
so human

File: src/main.rs [hunk 1]
```diff
@@ -315,11 +315,20 @@ async fn provide_interactive(
 
     let (db, hash) = provider::create_collection(sources).await?;
 
-    println!("Collection: {}\n", Blake3Cid::new(hash));
+    out_writer
```
Comment (dignifiedquire) [inline src/main.rs:318]: are you sure we want to push all this to stderr? seems to me that this is stdout content for the provider

Comment (flub) [reply src/main.rs:318]: Ugh, good catch.  I'm not sure at all.  I guess this is "data" and not progress information so should go to stdout?  But like the peer ID, auth token and ticket are also written using to OutWriter, and this seems to fall in that category to me and why I changed this.  Are they wrong too?

Also see, https://github.com/n0-computer/iroh/pull/781/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR319-R327

Comment (dignifiedquire) [reply src/main.rs:318]: I think they are wrong too, yes

Comment (flub) [reply src/main.rs]: Ok, updated them all to direct println! calls.  PTAL

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**